### ISSUE TITLE
Single EXE with an external appsettings.json

### ISSUE THREAD

**Issue by codemonkey85** [dotnet/designs#73]
Hello,

I was just wondering if there is a way or a plan to allow single .EXE deployment of my .NET Core apps, while still allowing for an external config file, i.e. appsettings.json.

I didn't see any options that get me to where I want to be: https://github.com/dotnet/designs/blob/master/accepted/single-file/design.md#user-experience

**Comment by jjxtra** [comment]
Having the same issue, except with app.config file. If I exclude them from the single file build, then I crash saying file not found. What we really need is a single file executable that does not extract itself to a temp/cache folder but instead runs right where it is, reading any config file in the same folder.

**Comment by zraffaele** [comment]
I've the same issue but only in i run the single EXE in windows service context

**Comment by jonasnordlund** [comment]
Yes, I was having the same problem in a .NET Core 3.1 project with `Microsoft.Extensions.Configuration` with single file executables. The solution might be to manually load the application configuration file with a `ConfigurationBuilder` and at the step where you tell it about the configuration file to load, specifiy the full path.

Now, the next question is "what if the full path is the temporary extraction directory?"

My tests has shown that `System.IO.Directory.GetCurrentDirectory()` reports the "expected" path for your config file reads -- as in the path where your executable is on the disk pre-extraction. While a call like `System.Reflection.Assembly.GetEntryAssembly().Location` reports the path to the temporary extraction folder for these single file exe's.
